src/spatial/collision.js

```diff
@@ -289,11 +281,11 @@ Crafty.c("Collision", {
             id = obj[0];
 
             //check if not added to hash and that actually intersects
-            if (!dupes[id] && this[0] !== id && obj.__c[comp] && overlap(oarea, area))
+            if (!dupes[id] && this[0] !== id && obj.__c[component] && overlap(oarea, area))
                 dupes[id] = obj;
         }
 
-        for (key in dupes) {
+        for (key in dupes) if (dupes.hasOwnProperty(key)) {
```

Comment:
This is very essential, I'm glad you asked.
This is good practice when developing JavaScript in general, but Crafty tries to avoid any unnecessary statements because of performance. It is even more essential here, because this method gets executed very often - probably every 20ms.
In this case we created the array ourselves, so we can be 100% sure there can't be any other properties there (unless some external code overwrites standard prototypes).
Revert please.